examples/solvers/chambolle_pock_preconditioner_denoising.py

```diff
@@ -0,0 +1,180 @@
+# Copyright 2014-2016 The ODL development group
+#
+# This file is part of ODL.
+#
+# ODL is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# ODL is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with ODL.  If not, see <http://www.gnu.org/licenses/>.
+
+"""Total variation base image denoising using the Chambolle-Pock solver.
+
+Let X and Y be finite-dimensional Hilbert spaces and K a linear mapping
+from X to Y with induce norm ||K||. The (primal) minimization problem we
+want to solve is
+
+    min_{x in X} F(K x) + G(x)
+
+where the proper, convex, lower-semicontinuous functionals
+F : Y -> [0, +infinity] and G : X -> [0, +infinity] are given
+by an l2-data fitting term regularized by isotropic total variation
+
+    F(K x) = 1/2 ||x - g||_2^2 + lam || |grad(x)| ||_1
+
+and by the indicator function for the set fo non-negative components of x
+
+   G(x) = {0 if x >=0, infinity if x < 0} ,
+
+respectively. Here, g denotes the image to denoise, ||.||_2 the l2-norm,
+||.||_1 the l1-semi-norm, grad the spatial gradient, lam the regularization
+parameter, |.| the point-wise magnitude across the vector components of
+grad(x), and K is a column vector of operators K = (id, grad)^T with
+identity mapping id.
+
+In order to use the Chambolle-Pock solver, we have to create the column
+operator K, choose a starting point x, create the proximal operator for G,
+create the proximal operator for the convex conjugate of F, choose the
+step sizes tau and sigma such that tau sigma ||K||_2^2 < 1, and set the
+total number of iterations.
+
+For details see :ref:`chambolle_pock`, :ref:`proximal_operators`, and
+references therein.
```

Comment:
Totally copy and paste. Shortening should probably be done in conjunction with moving this to the documentation.